assets/Script/technology/techGame1.js

```diff
@@ -0,0 +1,75 @@
+const { changeScene } = require('../util/sceneUtils');
+const { MENU_SCENE } = require('../constants');
+const MAX_BLOCK_NUMBER = 6;
+
+const generateNumberBlocks = () => {
+    return Math.floor(Math.random() * MAX_BLOCK_NUMBER);
+};
+
+const setupEventHandlers = (that) => {
+  that.homeSprite.on('mousedown', () => {
+    changeScene(MENU_SCENE);
+  });
+};
+
+cc.Class({
+    extends: cc.Component,
+
+    properties: {
+        speed : 200
+    },
+
+    onLoad () {
+        this.homeSprite = this.node.getChildByName('home_button');
+        this.blockSprite = this.node.getChildByName('block')
+
+        var numberBlocks = generateNumberBlocks();
+        var blockPosY = this.blockSprite.getPosition().y;
+        var blockWidth = this.blockSprite.getBoundingBox().size.width;   
+        
+        var distanceTwoBlocks = 30;
+
+        //total widths of all blocks
+        var totalWidth = blockWidth * numberBlocks  + distanceTwoBlocks * (numberBlocks - 1)
+        var blockPosX = - totalWidth / 2;
+
+        //first block is at the most left position
+        this.blockSprite.setPosition(blockPosX, blockPosY);
+        this.allBlocks = [this.blockSprite];
+        
+        for (i = 0; i < numberBlocks - 1; i++){

```

Comment:
Not declaring `i` causes this code to break 😢

Reply:
Thanks for pointing out. Actually, it did not break so I did not notice that. Will fix